pwndbg/dt.py

```diff
@@ -100,22 +100,29 @@ def dt(name='', addr=None, obj = None):
     if addr is not None:
         obj = pwndbg.memory.poi(t, addr)
 
-    # Header, optionally include the name
-    header = name
-    if obj: header = "%s @ %s" % (header, hex(int(obj.address)))
-    rv.append(header)
-
     if t.strip_typedefs().code == gdb.TYPE_CODE_ARRAY:
         return "Arrays not supported yet"
     if t.strip_typedefs().code not in (gdb.TYPE_CODE_STRUCT, gdb.TYPE_CODE_UNION):
         t = {name: obj or gdb.Value(0).cast(t)}
 
+    # Header, optionally include the name
+    header = name
+    if obj: header = "%s @ %s" % (header, hex(int(obj.address)))
+    rv.append(header)
+
+    endpos = 0
     for name, field in t.items():
         # Offset into the parent structure
-        o     = getattr(field, 'bitpos', 0)/8
+        o     = getattr(field, 'bitpos', 0)//8
```

Comment:
The `8` here is used like two times and is a magic number we can move into some constant e.g. `BITS_IN_BYTE = 8`.

Reply:
Meh.  The field is called `bitpos`, the constant is 8.  If you're concerned about future DEFCON 9-bit-byte, 27-bit-word type things, I don't plan to support them in Pwndbg 😛